src/storage/v3/vertex_accessor.cpp

```diff
@@ -237,6 +238,44 @@ Result<LabelId> VertexAccessor::PrimaryLabel(const View view) const {
   return vertex_->primary_label;
 }
 
+Result<PrimaryKey> VertexAccessor::PrimaryKey(const View view) const {
+  bool exists = true;
+  bool deleted = false;
+  Delta *delta = nullptr;
+  {
+    std::lock_guard<utils::SpinLock> guard(vertex_->lock);
+    deleted = vertex_->deleted;
+    delta = vertex_->delta;
```

Comment:
can we avoid acquiring this mutex?

Reply:
That mutex will be eliminated in [my PR](https://github.com/memgraph/memgraph/pull/460) completely.